api/team.go

```diff
@@ -70,7 +76,70 @@ func signupTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(model.MapToJson(m)))
 }
 
+func createTeamFromSSO(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	service := params["service"]
+
+	if !utils.IsServiceAllowed(service) {
+		c.SetInvalidParam("createTeamFromSSO", "service")
+		return
+	}
+
+	team := model.TeamFromJson(r.Body)
+
+	if team == nil {
+		c.SetInvalidParam("createTeamFromSSO", "team")
+		return
+	}
+
+	team.PreSave()
+
+	team.Name = model.CleanTeamName(team.Name)
+
+	if err := team.IsValid(); err != nil {
+		c.Err = err
+		return
+	}
+
+	team.Id = ""
+
+	found := true
+	count := 0
+	for found {
```

Comment:
I disagree that it's unclear, also I'd prefer to not create functions for one time use pieces of code. 0/5 though